src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,95 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static Object[] ARRAY;
+    private int capacity = 10;
+    private int size = 0;
+
+    public ArrayList() {
+        ARRAY = new Object[capacity];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            ARRAY = grow();
+        }
+        ARRAY[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            ARRAY = grow();
+        }
+        System.arraycopy(ARRAY, index, ARRAY, index + 1, size - index);
+        ARRAY[index] = value;
+        size++;
+    }
 
+    private Object[] grow() {
+        capacity = capacity + capacity / 2;
+        Object[] newArray = new Object[capacity];
+        System.arraycopy(ARRAY, 0, newArray, 0, size);
+        return newArray;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (!list.isEmpty()) {
+            add(list.remove(0));
```

Comment:
remove cost a lot for a list(you did it by yourself and know that); let's iterate through list and use method get()